### PR TITLE
Satellite support - port to Fedora, batch nr. 1

### PR DESCRIPTION
**Pull request by M4rtinK**
This is a very small initial batch that basically contains all commits related but not directly dependent on Satellite related changes. Still, we won't have to care about those in the later batches if we handle them now. :)

### CODE REVIEW

**Comment by M4rtinK** [comment]
Looking into the unit test & pylint issues. :)

**Comment by M4rtinK** [comment]
Looks like this is at least partially caused by not backporting the syspurpose DBus PR first - so lets do that #3690 and then revisit this one. I'll rebase my local fork in the meantime so we can get this PR going right away once the syspurpose PR is merged.

**Comment by jkonecny12** [comment]
Just write here a comment that it's prepared for review.

**Comment by M4rtinK** [comment]
> Just write here a comment that it's prepared for review.

Sure - it should be ready now, rebased on top of the master branch with the syspurpose PR merged. 

Tests are now also passing as far as I can tell. :)

**Review by jkonecny12 — APPROVED** [review]
Looks good to me. Thanks!

File: tests/unit_tests/pyanaconda_tests/modules/subscription/test_system_purpose.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    """Test the system purpose data handling code."""
+
+    def test_system_purpose_valid_json_parsing(self):
+        """Test that the JSON file holding valid system purpose values is parsed correctly."""
+        # check file missing completely
+        # - use path in tempdir to a file that has not been created and thus does not exist
+        with tempfile.TemporaryDirectory() as tempdir:
+            no_file = os.path.join(tempdir, "foo.json")
+            roles, slas, usage_types = get_valid_fields(valid_fields_file_path=no_file)
+            self.assertListEqual(roles, [])
```
Comment (poncovka) [inline tests/unit_tests/pyanaconda_tests/modules/subscription/test_system_purpose.py:60]: Please, drop this commit and create a new one with the upstream code. All `self.assert` calls were replaced with the `assert` statement.

**Review by poncovka — CHANGES_REQUESTED** [review]
You replace new code with the old one.

File: tests/unit_tests/pyanaconda_tests/modules/subscription/test_system_purpose.py (hunk 1 above)
Comment (M4rtinK) [reply tests/unit_tests/pyanaconda_tests/modules/subscription/test_system_purpose.py]: Ooops, now I see it. :P Should be fixed now. :)

**Review by poncovka — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**